yaml_edit/test/parse_test.dart

```diff
@@ -94,19 +94,28 @@ void main() {
     test('with the correct value', () {
       final doc = YamlEditor("YAML: YAML Ain't Markup Language");
 
-      expect(doc.parseAt(['YAML']).value, "YAML Ain't Markup Language");
+      expect(doc.parseAt(['YAML'])!.value, "YAML Ain't Markup Language");
     });
 
     test('with the correct value in nested collection', () {
       final doc = YamlEditor('''
 a: 1
-b: 
+b:
   d: 4
   e: [5, 6, 7]
 c: 3
 ''');
 
-      expect(doc.parseAt(['b', 'e', 2]).value, 7);
+      expect(doc.parseAt(['b', 'e', 2])!.value, 7);
+    });
+
+    test('with a null value in nested collection', () {
+      final doc = YamlEditor('''
+key1:
+  key2: null
+''');
+
+      expect(doc.parseAt(['key1', 'key2'])!.value, null);
     });
```

Comment:
Test for `parseAt()` with null value on a nested path.

Reply:
Let's make `parseAt` always return `YamlNode` instead of `YamlNode?` -- as you've suggested, then we don't need `!` all over the code.